text/0000-forbidden-value-trait.md

```diff
@@ -23,12 +23,13 @@ unsafe trait ForbiddenValues {
 }
 ```
 
-To implement `ForbiddenValues` for type `T`, the following condition must be met:
+To implement `ForbiddenValues` for type `T`, the following conditions must be met:
 
-For each value `t` of type `T`, and for each item `f` in `T::FORBIDDEN`:
+1. For each value `t` of type `T`, and for each item `f` in `T::FORBIDDEN`:
 ```rust
 *(&raw const t as *const [u8; size_of::<T>()]) != f
 ```
+2. `T` must not be a type that has forbidden values the compiler already knows about (e.g. `char`)
```

Comment:
> e.g. `char`

How about other types, are there any other types that the compiler already knows about? Is it possible to lists them out?

Reply:
Not sure how relevant this is now that the issue's closed, but https://doc.rust-lang.org/std/#primitives is a list of primitive types in the language. Which of these counts as "having forbidden values" depends on what you mean by that. For example, "uninitialized" is a forbidden / insta-UB-causing value for all of them, and `str` has to be valid UTF-8 within safe code but unsafe code can violate that without triggering UB, and I don't know if you'd want to include library types like `NonZero*`, and so on and so forth. But if we try to just ignore all of this scope vagueness, I think `bool` is the only other primitive type that is "like `char`" insofar as it clearly has to be implemented as one of the integer types with certain values excluded (in `bool`'s case, every value except 0 or 1).